structures/radix_sort.py

```diff
@@ -0,0 +1,82 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+
+def radix_sort(iterable, num_base=10):
+    max_length = False
+    tmp, decimal_place = -1, 1
+
+    while not max_length:
+        max_length = True
+
+        buckets = [[] for x in xrange(num_base)]
+
+        for item in iterable:
+            tmp = item // decimal_place
+            buckets[tmp % num_base].append(item)
```

Comment:
We kind of have the illusion that this will work with bases other than 10, but unless you manufacture a set of int objects that appear to be octals or hexidecimals, this approach to the radix sort will not work.  Try it on a series of octal numbers.